src/emulator/io/src/io.cpp

```diff
@@ -357,30 +257,28 @@ int write_file(SceUID fd, const void *data, SceSize size, const IOState &io, con
             LOG_INFO("*** TTY: {}", s);
             return size;
         }
-
         return IO_ERROR_UNK();
     }
 
-    LOG_TRACE("{}: Writing file: fd: {}, size: {}", export_name, log_hex(fd), size);
-
-    const StdFiles::const_iterator file = io.std_files.find(fd);
-    if (file != io.std_files.end()) {
-        return static_cast<int>(fwrite(data, 1, size, file->second.file_handle.get()));
+    const auto file = io.std_files.find(fd);
+    if (file->second.can_write_file()) {
+        const auto written = file->second.write(data, 1, size);
+        LOG_TRACE("{}: Writing to fd: {}, size: {}", export_name, log_hex(fd), size);
+        return static_cast<int>(written);
     }
 
     return IO_ERROR(SCE_ERROR_ERRNO_EBADFD);
 }
 
-SceOff seek_file(SceUID fd, SceOff offset, int whence, IOState &io, const char *export_name) {
-    assert((whence == SCE_SEEK_SET) || (whence == SCE_SEEK_CUR) || (whence == SCE_SEEK_END));
```

Comment:
Quite sure when whence is incorrect, a different error than SCE_ERROR_ERRNO_EBADFD is returned.